### PR TITLE
Correct Inconsistencies in GCP, AWS, and Azure Reference Docs

### LINKED ISSUE #1146: Standardize `Default cluster` values for GCP + AWS + Azure
This list is not exhaustive, but I noticed a few discrepancies between these docs across Clouds:

- Is the metadata for each doc the same?
- The AWS doc links out to AWS documentation for some of the resources. Should we do that for GCP and Azure too?
- The Azure doc does not include worker queue related resources -- e.g. `Worker node pool`, `Airflow system node pool`
- Add periods `.` to the end of all descriptions?
- Keep order and descriptions of each resource consistent -- e.g. `S3 bucket for storage of Airflow task logs.` for AWS vs `Stores Airflow task logs` for Azure.

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1146 

@dnarain-astro confirmed that _Worker node pool_ and _Airflow system node pool_ entries should be added to the  _Default cluster values_ tables in the topics topic [Resources required for Astro on Azure](https://docs.astronomer.io/astro/resource-reference-azure) and[ Resources required for Astro on GCP](https://docs.astronomer.io/astro/resource-reference-gcp),

@paolaperaza when time permits, please identify any additional changes you think are necessary for consistency. Thanks!

### CODE REVIEW

**Comment by iamakashladdha** [comment]
@paolaperaza 

Please find comments per review of these documents mainly with respect to worker node pools, Astro node pools and airflow system node pools.

- It is mentioned **Astro System Node Pool** for AWS, while for GCP as **Astro node pool.**, Same for **Airflow system node pool** for AWS and **Airflow node pool** for GCP, it should be in sync along with the Description of this

- Azure has worker node pool and Airflow system node pool description as is copied from AWS, while astro node pool is not mentioned for Azure.

- Do you want to mention the Min and Max configured nodes for Airflow System Node Pool and Astro System Node pool to provide better transparency to customers? 

- It is mentioned that Airflow System node pool and Astro System node pool **are fully managed by Astronomer**, these can raise questions for the worker node pools, Astro is a fully managed offering so worker node pools also should be fully managed by Astronomer. Maybe you would want to include another statement to lay emphasis that node size and node scaling configurations are fully taken care of by Astronomer for these 2 node pools

Further observations are below. 

For **astro/resource-reference-gcp.md**

1. Incorrect Full Stop in the first line **Virtual private network for hosting GCP. resources** for Resource VPC.
2. Worker Node Pool and Airflow System Node pool blocks are repeated, also the newly added ones look like they are copied as is from AWS as in Snap below. 
image

3. For Maximum Node Count - **The maximum number of nodes that your Astro cluster can support.** -As this only applies to worker node pools, can we mention The maximum number of **worker** nodes that your Astro cluster can support? **Also for Azure and AWS**
4. Worker node pools - For Description it should be **Worker node pools runs all Airflow workers** or **Node pools that run all Airflow workers** instead of Node pools run all Airflow workers. 
5.  Similar change is needed for the Astro Node pool and Airflow node pool as mentioned in #4.
6. For Astro System Node pools (data plane nodes) we should mention that nodes are created per AZ and also the number of AZ that are used.

For **astro/resource-reference-azure.md**

1. Worker node pool and Airflow system Node pool are again copied from AWS even with the same machine types and references for EC2 instances. Astro node pool is not mentioned, All the node pools need to be mentioned with respective machine types.
image

2. Virtual Machines (nodes) - Should be removed and we follow the same terminologies of Astro pool, Worker Node pool and Airflow system Node pool across the 3 cloud providers.
3. For Astro System Node pools (data plane nodes) we should mention that nodes are created per AZ and also the number of AZ that are used. **Same for AWS too.**

Regards, @iamakashladdha

**Comment by kwcanuck** [comment]
@iamakashladdha thank you for your through review and feedback. I have done my best to incorporate the changes you requested. Unfortunately, some of the changes you requested are unclear. For example, I'm not sure what you mean by "Similar change is needed for the Astro Node pool and Airflow node pool as mentioned in https://github.com/astronomer/docs/issues/4" and "...that nodes are created per AZ". What is AZ?

It's often easier if you edit the existing text directly. To do this:

1. Select the **Files changed** tab in this PR.

2. Click the plus symbol (+) next to a line of text you want updated or where new text should be inserted.

image

3. To edit existing text, click the **Add a suggestion icon** and then add your edits.

image

4. To add new text, type the text in the dialog.

5. Click **Add single comment**.

6. When you've completed your review, click **Review changes**, select an option, and then click **Submit review**.

image

**Review by dnarain-astro — APPROVED** [review]

**Comment by kwcanuck** [comment]
@iamakashladdha I apologize, but I am having difficulty determining what changes are required to these topics. Can you provide your edits as I recommended?

File: astro/resource-reference-azure.md [hunk 1]
```diff
@@ -12,18 +12,21 @@ Unless otherwise specified, new clusters on Microsoft Azure are created with a s
...
+| [Private DNS Zone for Blob Storage](https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns)             | Provides access to Azure Blob storage task logs.                                                                                              | 1x                                                                                            |
 | Public IP Address                             | Required for connectivity to the control plane and other services.                                                                            | 1x                                                                                            |
+| Maximum Node Count | The maximum number of worker nodes that your Astro cluster can support. When this limit is reached, your Astro cluster can't auto-scale and worker Pods may fail to schedule. | 20 |
```
Comment (paolaperaza) [inline astro/resource-reference-azure.md:29]: Edit to fit AWS description of max node count

File: astro/resource-reference-gcp.md [hunk 2]
```diff
@@ -15,17 +15,17 @@ Read the following document for a reference of our default resources as well as
...
+| Airflow node pool | A node pool runs all core Airflow components such as the scheduler and webserver. This node pool is fully managed by Astronomer. | 1x pool of n2-standard-4 nodes |
+| [Service Network Peering](https://cloud.google.com/vpc/docs/configure-private-services-access) | The Astro VPC is peered to the Google Service Networking VPC.                                         | 1, IP Range is `172.23.0.0/19` |
+| [NAT Router (External)](https://cloud.google.com/nat/docs/overview)   | Required for connectivity with the Astro control plane and other public services                     | 1.                            |
 | [GKE Cluster](https://cloud.google.com/kubernetes-engine/docs/concepts/kubernetes-engine-overview)             | A GKE cluster is required to run the Astro data plane, which hosts the resources and data required to execute Airflow tasks. Workload Identity is enabled on this cluster. | 1x, IP Ranges are `172.21.0.0/19` for cluster IPs and `172.22.0.0/19` for cluster services |
```
Comment (paolaperaza) [inline astro/resource-reference-gcp.md:24]: Same -- bring this up to the top

File: astro/resource-reference-gcp.md [hunk 3]
```diff
@@ -15,17 +15,17 @@ Read the following document for a reference of our default resources as well as
...
-| Maximum Node Count | The maximum number of nodes that your Astro cluster can support. When this limit is reached, your Astro cluster can't auto-scale and worker Pods may fail to schedule. | 20 |
+| [Workload Identity Pool](https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers) | Astro uses the fixed Workload Identity Pool for your project. One is created if it does not exist. | The default pool (`PROJECT_ID.svc.id.goog`) is used |
+| [Cloud SQL for PostgreSQL](https://cloud.google.com/sql/docs/postgres) | The Cloud SQL instance is the primary database for the Astro data plane. It hosts the metadata database for each Airflow Deployment hosted on the GKE cluster. | 1x regional instance with 4 vCPUs, 16GB memory |
+| [Google Cloud Storage (GCS) Bucket](https://cloud.google.com/storage/docs/creating-buckets) | GCS bucket to store Airflow task logs. | 1 bucket with name `airflow-logs-<clusterid>` |
+| Maximum Node Count | The maximum number of worker nodes that your Astro cluster can support. When this limit is reached, your Astro cluster can't auto-scale and worker Pods may fail to schedule. | 20 |
```
Comment (paolaperaza) [inline astro/resource-reference-gcp.md:28]: Same -- edit value

**Review by paolaperaza — APPROVED** [review]
Giving you a ✅ here Rob! A few comments but we can iterate over time

File: astro/resource-reference-azure.md [hunk 4]
```diff
@@ -12,18 +12,21 @@ Unless otherwise specified, new clusters on Microsoft Azure are created with a s
...
+| [Resource Group](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal)                                | A container for cluster resources.                                                                                                            | 1x                                                                                            |
+| Worker node pool | A node pool that runs all workers across Deployments in the cluster. The number of nodes in the pool auto-scales based on the demand for workers in your cluster. You can configure multiple worker node pools to run tasks on different worker types. This node pool is fully managed by Astronomer. | 1x pool of m5.xlarge nodes |
+| Airflow node pool | A node pool that runs all core Airflow components, including the scheduler and webserver. This node pool is fully managed by Astronomer. | 1x pool of m5.xlarge nodes |
```
Comment (paolaperaza) [inline astro/resource-reference-azure.md:17]: Ope -- wait, these can't be right. `m5.xlarge` is an AWS only term. We need to ask infra what node types these are for Azure.

File: astro/resource-reference-azure.md [hunk 5]
```diff
@@ -12,18 +12,21 @@ Unless otherwise specified, new clusters on Microsoft Azure are created with a s
...
-| Azure Kubernetes Service (AKS) Cluster        | Runs the Astro Data Plane, which hosts the resources and data required to execute Airflow tasks.                                              | 1x                                                                                            |
-| Virtual Machines (nodes)                      | Hosts all system and Airflow components on Astro, including workers and schedulers. Auto-scale based on the demand for nodes in your cluster. | Standard_D4d_v5                                                                               |
+| [Azure Kubernetes Service (AKS) Cluster](https://docs.microsoft.com/en-us/azure/aks/intro-kubernetes)        | Runs the Astro Data Plane, which hosts the resources and data required to execute Airflow tasks.                                              | 1x                                                                                            |
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/resource-reference-azure.md:23]: @paolaperaza sorry, bring what to the top? Can you clarify?

File: astro/resource-reference-azure.md (hunk 4 above)
Comment (kwcanuck) [reply astro/resource-reference-azure.md:18]: @paolaperaza I confirmed with @dnarain-astro that these values should be Standard_D4d_v5. The doc has been updated.

File: astro/resource-reference-azure.md [hunk 6]
```diff
@@ -12,18 +12,21 @@ Unless otherwise specified, new clusters on Microsoft Azure are created with a s
...
+| [Resource Group](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal)                                | A container for cluster resources.                                                                                                            | 1x                                                                                            |
+| Worker node pool | A node pool that runs all workers across Deployments in the cluster. The number of nodes in the pool auto-scales based on the demand for workers in your cluster. You can configure multiple worker node pools to run tasks on different worker types. This node pool is fully managed by Astronomer. | 1x pool of m5.xlarge nodes |
+| Airflow node pool | A node pool that runs all core Airflow components, including the scheduler and webserver. This node pool is fully managed by Astronomer. | 1x pool of m5.xlarge nodes |
+| Astro node pool | A node pool that runs all other system components required in Astro. The availability zone determines how many nodes are created. This node pool is fully managed by Astronomer.| 1x pool of m5.xlarge nodes |
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/resource-reference-azure.md]: The update has been implemented as requested.

**Pull request merged**